src/drivers/driver/src/lib.rs

```diff
@@ -1,46 +1,165 @@
 #![no_std]
 #![allow(non_snake_case)]
 
+use core::slice::{from_raw_parts, from_raw_parts_mut};
+
+pub type Result<T> = core::result::Result<T, &'static str>;
+
+const EOF : Result<usize> = Err("EOF");
+const NOT_IMPLEMENTED : Result<usize> = Err("not implemented");
+
 pub trait Driver {
     /// Initialize the driver.
-    fn init(&self);
+    fn init(&mut self);
     /// Returns number of bytes read.
-    fn read(&self, data: &mut [u8]) -> usize;
+    fn pread(&self, data: &mut [u8], pos: usize) -> Result<usize>;
```

Comment:
so we're not doing the thing where the pread returns the amount and the "your next read would be with this offset?"

Reply:
I found this works well for incrementing the cursor:
```
cursor += drv.pread(data, cursor)?;
```